subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultObjectConfigurationAction.java

```diff
@@ -97,9 +102,14 @@ private void applyScript(Object script) {
         ScriptSource scriptSource = new UriScriptSource("script", scriptUri);
         ClassLoaderScope classLoaderScopeChild = classLoaderScope.createChild("script-" + scriptUri.toString());
         ScriptHandler scriptHandler = scriptHandlerFactory.create(scriptSource, classLoaderScopeChild);
-        ScriptPlugin configurer = configurerFactory.create(scriptSource, scriptHandler, classLoaderScopeChild, classLoaderScope, false);
-        for (Object target : targets) {
-            configurer.apply(target);
+        final ScriptPlugin configurer = configurerFactory.create(scriptSource, scriptHandler, classLoaderScopeChild, classLoaderScope, false);
```

Comment:
Yeah I thought about that. My rationale for this first iteration is that only the actual 'application' of the scripts is taken into account. But we might want to include classloader scope, script handler and script plugin creations.

In this particular case we are applying to a set of targets. We need a build operation for each, only the first one would then account for the setup.

@adammurdoch, do you concur?